Simplified/NYPLBookRegistryRecord.m

```diff
@@ -79,13 +79,7 @@ - (instancetype)initWithBook:(NYPLBook *const)book
 
   if (!actuallyOnHold) {
     // Set the correct non-holding state.
-    if (!((NYPLBookStateDownloadFailed |
-           NYPLBookStateDownloading |
-           NYPLBookStateDownloadNeeded |
-           NYPLBookStateDownloadSuccessful |
-           NYPLBookStateUsed)
-          & self.state)
-        && self.state != NYPLBookStateUnregistered)
+    if (self.state && (self.state == NYPLBookStateHolding || self.state == NYPLBookStateUnsupported))
```

Comment:
`self.state &&` is not needed here because it's essentially an integer, which in objc evaluates to false if it's 0 and true in all other cases.